test/integration/kubernetes_deploy_test.rb

```diff
@@ -616,6 +616,42 @@ def test_output_when_switching_labels_incorrectly
     ], in_order: true)
   end
 
+  def test_scale_down_deployment
+    dep_name = nil
+    pod_name = nil
+    pod_status = "Running"
+    source = 3
+    # Create a deployement with 3 pods
+    result = deploy_fixtures("hello-cloud", subset: ["configmap-data.yml", "web.yml.erb"]) do |fixtures|
+      web = fixtures["web.yml.erb"]["Deployment"].first
+      web["spec"]["replicas"] = source
+      # Disable grace period in deleting pods to speed up test
+      web["spec"]["template"]["spec"]["terminationGracePeriodSeconds"] = 0
+      dep_name = web["metadata"]["name"]
+      pod_name = web["spec"]["template"]["metadata"]["labels"]["name"]
+    end
+    assert_deploy_success(result)
+    hello_cloud = FixtureSetAssertions::HelloCloud.new(@namespace)
+    hello_cloud.assert_pod_status(pod_name, pod_status, 3)
+    assert_logs_match_all([
+      %r{Service/web\s+Selects at least 1 pod},
+      %r{Deployment/web\s+3 replicas, 3 updatedReplicas, 3 availableReplicas}
+    ])
+    # Scale down to 1 pod
```

Comment:
Why not do a second deploy here with `deploy_fixtures`, setting `web["spec"]["replicas"] = 0`? The problem was that a deploy that scaled replicas to zero would give the wrong output (claimed the deploy timed out), which we're not really able to test if the scaling is done externally. Looking back at the issue, it looks possible that it was actually another incarnation of the instant "timeout" bug we fixed two weeks ago, so if you're now unable to replicate the output pasted in the issue, that could be why.

Reply:
Just updated the test code to use deploy_fixtures to apply the change of replicas instead of calling kubectl scale directly.

I couldn't reproduce the logs originally posted in the issue.
I ran test test_can_deploy_deployment_with_zero_replicas in current master and in a commit back to Sept 28, but still couldn't reproduce it. 
Do you still remember about the timeout issue that was fixed 2 weeks ago? I can pull out the specific commit and try it out again. Was the problem logs generated consistently in CI only or locally as well?

Reply:
> I ran test test_can_deploy_deployment_with_zero_replicas in current master and in a commit back to Sept 28, but still couldn't reproduce it.

That test doesn't reproduce the issue; it's something we saw in production (the screenshot is of Shipit logs). I don't think anyone had a chance to try to reproduce locally or with a test. We opened the issue to remind ourselves to try that, but you're the first to pick it up. 😄 

This is the PR that fixed the timeout issue I mentioned: https://github.com/Shopify/kubernetes-deploy/pull/213. One approach would be to check out  a commit from the beginning of August, when the issue you're investigating was filed, and make sure your new test reproduces the bug (i.e. fails) against that code. If it doesn't, please ping me and we can decide what to do about this. I'm not sure the bug from #213 was the cause, but if it was, it would make the issue nearly impossible to reproduce reliably.